### PR TITLE
changed login logic and added account methods

### PR DESCRIPTION
**Pull request by BerryLiu2002**
### Summary
Changed login logic:
removed circular references with conn storing the account
Login logic changed to:
- create Account (acc)
- call acc.connection.login()
- use acc.connection.session obj

Added assignment methods
- get_assignment_submissions
- get_assignments
- get_assignment_submission

### Details
NOTE: one text not passing -> extensions test_get_extensions

### Checks
- [x] Tested changes
- [ ] Attached Logs

### Team to Review
_Mention the name of the team to review the PR_

### Reference to the issue
_Mention the issue ID or resources_

### CODE REVIEW

**Comment by BerryLiu2002** [comment]
Do the tests run without the .env file?

**Comment by KevZhe** [comment]
> Do the tests run without the .env file? I ran them locally with the .env file and they passed

Yea, they read in the environment variables using GitHub Secrets. Will try look into it

File: src/sylveon/_classes/account.py
Comment (KevZhe) [inline src/sylveon/_classes/account.py:1]: wait, are there two account files? 
- _account.py and account.py
was this intentional?

**Comment by KevZhe** [comment]
just a general comment, but splitting the changes into a couple different commits would make it easier to review

File: src/sylveon/_classes/account.py
Comment (BerryLiu2002) [reply src/sylveon/_classes/account.py:1]: just the _account one, this other one is a mistake

**Comment by calvinatian** [comment]
> > Do the tests run without the .env file? I ran them locally with the .env file and they passed
> 
> Yea, they read in the environment variables using GitHub Secrets. Will try look into it

The test is failing because it doesn't have access to the repository secrets. Workflows triggered from a fork don't get passed secret values as a security precaution (on a public repo someone could fork the repo and open a PR that reads the repository secrets and sends them somewhere).

The easiest solution is probably to just push your changes to a branch on this repo instead of your fork. Then open a PR from your branch to main.

As long as the tests are passing locally another option is to just ignore the tests here and merge anyways. Once merged to main it should trigger another CI run.

**Pull request closed without merging**